download.py

```diff
@@ -136,3 +72,169 @@ def download_run_aspc(accession, term, out):
         logging.info("Current Run: %s has been successfully downloaded", accession)
         return True
     return False
+
+
+method_to_download_function = {
+    "f": download_run_ftp,
+    "a": download_run_aspc,
+    "q": download_run_fasterq_dump,
+}
+
+
+def _make_accession_list(term_: str) -> list[str]:
+    """Get an accession list based on pattern of the given term."""
+    if SRR_PATTERN.search(term_):
+        accession_list = [term_]
+    elif SRP_PATTERN.search(term_):
+        accession_list = metadata.get_srr_ids_from_srp(term_)
+    else:
+        raise ValueError(f"Unknown pattern: {term_}")
+
+    return accession_list
+
+
+def handle_methods(term_: str, method_: str, out) -> bool:
```

Comment:
Why the underscores in arg names?

Reply:
@dakq thanks for asking. Otherwise we would have variables with the same name in the global and local scope. We already have `term` in the global scope.

Reply:
> Otherwise we would have variables with the same name in the global and local scope

Is that a problem?

Reply:
@dakq not really. I renamed them